x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForRolloverReadyStep.java

```diff
@@ -53,6 +55,12 @@ public void evaluateCondition(IndexMetaData indexMetaData, Listener listener) {
             return;
         }
 
+        if (indexMetaData.getSettings().getAsBoolean(LIFECYCLE_ROLLOVER_SKIP_ROLLED, true) &&
```

Comment:
I think we'll also need this check in `RolloverStep`, or it'll just try to issue the rollover request there and fail a step later. I think `UpdateRolloverLifecycleDateStep`is fine not checking it, because we should probably update the reference date anyway, and the step to set indexing complete is probably good to do anyway.

Reply:
+1, otherwise we pass the check step but fail on the "actual" rollover

Reply:
Ah, I wasn't seeing this in the manual tests as we need to complete this step here (call the `listener` callback). I'll add the check to the `RolloverStep` too with the `listener` callback as well. Will capture this in an integration test too as you guys suggested